### PR TITLE
Explicit listing of console scripts

### LINKED ISSUE #2757: ImportError: cannot import name 'convert' when executing sct_utils
### Description 

Just by executing `sct_utils` got the following error message instead of getting the usual help message:
```
Traceback (most recent call last):
  File "/home/shernand/sct_4.3/scripts/sct_utils.py", line 28, in <module>
    from sct_convert import convert
  File "/home/shernand/sct_4.3/scripts/sct_convert.py", line 23, in <module>
    import sct_utils as sct
  File "/home/shernand/sct_4.3/scripts/sct_utils.py", line 28, in <module>
    from sct_convert import convert
ImportError: cannot import name 'convert'

```
### Steps to Reproduce

0. Installed SCT 4.3 on Ubuntu 20.04.
Output of `sct_check_dependencies`:

```
--
Spinal Cord Toolbox (4.3)

SCT info:
- version: 4.3
- path: /home/shernand/sct_4.3
OS: linux (Linux-5.4.0-37-generic-x86_64-with-debian-bullseye-sid)
CPU cores: Available: 8, Used by SCT: 8
RAM: MemTotal:       32554456 kB
total        used        free      shared  buff/cache   available
Mem:          31791        6319        2826         288       22645       24750
Swap:         11442           3       11439
Check Python executable.............................[OK]
  Using bundled python 3.6.10 |Anaconda, Inc.| (default, May  8 2020, 02:54:21) 
[GCC 7.3.0] at /home/shernand/sct_4.3/python/envs/venv_sct/bin/python
Check if data are installed.........................[OK]
Check if numpy is installed.........................[OK] (1.18.5)
Check if colored is installed.......................[OK] (1.4.2)
Check if dipy is installed..........................[OK] (1.1.1)
Check if futures is installed.......................[OK]
Check if h5py is installed..........................[OK] (2.10.0)
Check if ivadomed (1.2.1) is installed..............[OK] (1.2.1)
Check if Keras (2.1.5) is installed.................[OK] (2.1.5)
Check if matplotlib is installed....................[OK] (3.2.1)
Check if nibabel is installed.......................[OK] (3.1.0)
Check if pandas is installed........................[OK] (1.0.4)
Check if pyqt5 (5.11.3) is installed................[OK] (5.11.3)
Check if pytest is installed........................[OK] (5.4.3)
Check if pytest-cov is installed....................[OK] (2.9.0)
Check if raven is installed.........................[OK]
Check if requests is installed......................[OK] (2.23.0)
Check if requirements-parser is installed...........[OK] (0.2.0)
Check if scipy is installed.........................[OK] (1.4.1)
Check if scikit-image is installed..................[OK] (0.17.2)
Check if scikit-learn is installed..................[OK] (0.23.1)
Check if tensorflow (1.5.0) is installed............[OK] (1.5.0)
Check if torch (1.5.0+cpu) is installed.............[OK] (1.5.0+cpu)
Check if torchvision (0.6.0+cpu) is installed.......[OK] (0.6.0+cpu)
Check if xlwt is installed..........................[OK] (1.3.0)
Check if tqdm is installed..........................[OK] (4.46.1)
Check if transforms3d is installed..................[OK] (0.3.1)
Check if urllib3 is installed.......................[OK] (1.25.9)
Check if spinalcordtoolbox is installed.............[OK]
Check ANTs compatibility with OS ...................[OK]
Check PropSeg compatibility with OS ................[OK]
Check if DISPLAY variable is set....................[OK]
Check if figure can be opened with PyQt.............[OK]
```

1. Just executing `sct_utils`

**Expected behavior:** The usage/help message

**Actual behavior:** Got this import error
```
Traceback (most recent call last):
  File "/home/shernand/sct_4.3/scripts/sct_utils.py", line 28, in <module>
    from sct_convert import convert
  File "/home/shernand/sct_4.3/scripts/sct_convert.py", line 23, in <module>
    import sct_utils as sct
  File "/home/shernand/sct_4.3/scripts/sct_utils.py", line 28, in <module>
    from sct_convert import convert
ImportError: cannot import name 'convert'

```

### PR DESCRIPTION
**Pull request by saulo-p**
This PR fixes #2757 and introduces two changes:

- It removes the circular dependency between `sct_utils` and `sct_convert` to prevent the `ImportError`. (This change is a quick fix and should later turn into a refactoring of `sct_utils` out of the `scripts` directory).
- It changes the `setup.py` script to explicitly list the console scripts to be installed into the `bin` directory. 
This second change was motivated by the unintended usage of `sct_utils` as a script in the user story of the issue.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -62,11 +62,70 @@
...
+            'sct_deepseg_sc',
+            'sct_denoising_onlm',
+            'sct_detect_pmj',
+            'sct_dice_coefficient',
+            'sct_dmri_compute_bvalue',
+            'sct_dmri_compute_dti',
+            'sct_dmri_concat_b0_and_dwi',
+            'sct_dmri_concat_bvals',
+            'sct_dmri_concat_bvecs',
+            'sct_dmri_create_noisemask',
```
Comment (jcohenadad) [inline setup.py:99]: this one might be deprecated (when i launch it, nothing happens), so we might want to remove it

File: setup.py [hunk 2]
```diff
@@ -62,11 +62,70 @@
...
+            'sct_fmri_compute_tsnr',
+            'sct_fmri_moco',
+            'sct_get_centerline',
+            'sct_image',
+            'sct_label_utils',
+            'sct_label_vertebrae',
+            'sct_maths',
+            'sct_merge_images',
+            'sct_pipeline',
+            'sct_pipeline_makefig',
```
Comment (jcohenadad) [inline setup.py:116]: we could probably remove them (replaced by sct_run_batch)

File: setup.py [hunk 3]
```diff
@@ -62,11 +62,70 @@
...
-      #'sct_deepseg_gm=spinalcordtoolbox.deepseg_gm.__main__:main',
-     ],
+        '{}=spinalcordtoolbox.compat.launcher:main'.format(x) for x in \
+        [
+            'isct_check_detection',
+            'isct_convert_binary_to_trilinear',
+            'isct_minc2volume-viewer',
+            'isct_test_ants',
+            'isct_warpmovie_generator',
+            'msct_multiatlas_seg',
```
Comment (jcohenadad) [inline setup.py:72]: deprecated (used by sct_segment_graymatter, which is deprecated)

Comment (jcohenadad) [reply setup.py:72]: but we might want to revive this code at some point, so maybe we can leave the scripts, and just not expose them

Comment (Drulex) [reply setup.py]: deleted reference, kept script

File: setup.py (hunk 2 above)
Comment (Drulex) [reply setup.py]: Delete reference and script. Added `sct_run_batch` reference.

File: setup.py (hunk 1 above)
Comment (Drulex) [reply setup.py]: Deleted reference and script

**Comment by Drulex** [comment]
As per sack thread we are removing the deprecated functions and relying on the git history if we ever need to revisit old implementations.

@jcohenadad should be ready for final review.

**Pull request merged**